### ISSUE TITLE
PHP interface enhancements

### ISSUE THREAD

**Issue by garak** [Komodo/KomodoEdit#2335]
### Short Summary

a class that implements an interface lacks basic features, like validation and "inheritance" (quoted, because it's not a real inheritance, see below)

### Steps to Reproduce

```php
<?php

interface intA
{
    public function foo();
    public function bar();
}

class clsA implements intA
{
// ...
}
```

### Expected results

I expect two features:
* autocompletion of methods needed by interface. So, for example, if I start typing "public function f" I should be suggested with "foo", that is a method required by implemented interface.
* validation of missing method. The class should be marked as invalid until I don't write all the required methods

### Actual results

None

### Platform Information

*Komodo Edit or IDE?* IDE
*Komodo Version?* 10.1.4
*Operating System (and version)?* Any (it's Ubuntu 16.04 in my case)

### Additional Information

none

**Comment by mitchell-as** [comment]
Hi, thanks for the suggestions.

The first sounds reasonable, but the second depends on a linting tool that provides the method checking. If you can point us in the direction of a PHP linter that does that, it would help.

**Comment by garak** [comment]
You can use the php interpreter itself. If you save the above example as example.php and try to execute via command line, e.g. `php example.php` you'll get the following:
```
PHP Fatal error:  Class clsA contains 2 abstract methods and must therefore be declared abstract or implement the remaining methods (intA::foo, intA::bar) in example.php on line 9
```

**Comment by mitchell-as** [comment]
No, we certainly do not want to execute PHP files for syntax checking. That is very dangerous. It has to be static analysis.
